Header    header
MsgState  head
MsgState  abdomen
MsgState  left
MsgState  right
MsgState  aux
